### PR TITLE
[FLINK-16140] [docs-zh] Translate Event Processing (CEP) page into Chinese

### PR DESCRIPTION
**Pull request by shuai-xu**
## What is the purpose of the change

*This pull request translates the CEP page into Chinese.*

## Verifying this change

This change is a docs work without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: docs/dev/libs/cep.zh.md [hunk 1]
```diff
@@ -23,23 +23,20 @@ specific language governing permissions and limitations
...
 
 * This will be replaced by the TOC
 {:toc}
 
-## Getting Started
+## 开始
 
-If you want to jump right in, [set up a Flink program]({{ site.baseurl }}/dev/projectsetup/dependencies.html) and
-add the FlinkCEP dependency to the `pom.xml` of your project.
+如果你想现在开始尝试，[创建一个Flink程序]({{ site.baseurl }}/dev/projectsetup/dependencies.html)，
```
Comment (libenchao) [inline docs/dev/libs/cep.zh.md]: Add `/zh` to the link.

File: docs/dev/libs/cep.zh.md [hunk 2]
```diff
@@ -63,13 +60,12 @@ add the FlinkCEP dependency to the `pom.xml` of your project.
 </div>
 </div>
 
-{% info %} FlinkCEP is not part of the binary distribution. See how to link with it for cluster execution [here]({{site.baseurl}}/dev/projectsetup/dependencies.html).
+{% info %} FlinkCEP不是二进制分发的一部分。在集群上执行如何链接它可以看[这里]({{site.baseurl}}/dev/projectsetup/dependencies.html)。
```
Comment (libenchao) [inline docs/dev/libs/cep.zh.md]: how about "二进制发行版” ？

File: docs/dev/libs/cep.zh.md (hunk 1 above)
Comment (libenchao) [reply docs/dev/libs/cep.zh.md]: +n to all other links

File: docs/dev/libs/cep.zh.md [hunk 3]
```diff
@@ -63,13 +60,12 @@ add the FlinkCEP dependency to the `pom.xml` of your project.
...
 
-Now you can start writing your first CEP program using the Pattern API.
+现在可以开始使用Pattern API写你的第一个CEP程序了。
 
-{% warn Attention %} The events in the `DataStream` to which
-you want to apply pattern matching must implement proper `equals()` and `hashCode()` methods
-because FlinkCEP uses them for comparing and matching events.
+{% warn Attention %} `DataStream`中的事件，如果你想在上面进行模式匹配的话，必须实现合适的 `equals()`和`hashCode()`方法，
```
Comment (libenchao) [inline docs/dev/libs/cep.zh.md]: Attention -> 注意

File: docs/dev/libs/cep.zh.md [hunk 4]
```diff
@@ -136,140 +132,143 @@ val result: DataStream[Alert] = patternStream.process(
...
  </div>
 
-#### Conditions
+#### 条件
 
-For every pattern you can specify a condition that an incoming event has to meet in order to be "accepted" into the pattern e.g. its value should be larger than 5,
-or larger than the average value of the previously accepted events.
-You can specify conditions on the event properties via the `pattern.where()`, `pattern.or()` or `pattern.until()` methods.
-These can be either `IterativeCondition`s or `SimpleCondition`s.
+对每个模式你可以指定一个条件来决定一个进来的事件是否被接受进入模这个式，例如，它的value字段应该大于5，或者大于前面接受的事件的平均值。
```
Comment (libenchao) [inline docs/dev/libs/cep.zh.md]: 是否被接受进入模这个式 -> 是否被接受进入这个模式

File: docs/dev/libs/cep.zh.md [hunk 5]
```diff
@@ -665,12 +647,11 @@ pattern.oneOrMore().greedy()
 </div>
 </div>
 
-### Combining Patterns
+### 组合模式
 
-Now that you've seen what an individual pattern can look like, it is time to see how to combine them
-into a full pattern sequence.
+现在你已经看到单个的模式是什么样的了，改取看看如何把它们连接起来组成一个完整的模式序列。
```
Comment (libenchao) [inline docs/dev/libs/cep.zh.md]: 改取 -> 该去

File: docs/dev/libs/cep.zh.md (hunk 4 above)
Comment (shuai-xu) [reply docs/dev/libs/cep.zh.md]: done

File: docs/dev/libs/cep.zh.md (hunk 5 above)
Comment (shuai-xu) [reply docs/dev/libs/cep.zh.md]: done

File: docs/dev/libs/cep.zh.md (hunk 1 above)
Comment (shuai-xu) [reply docs/dev/libs/cep.zh.md]: done

File: docs/dev/libs/cep.zh.md (hunk 3 above)
Comment (shuai-xu) [reply docs/dev/libs/cep.zh.md]: done

File: docs/dev/libs/cep.zh.md [hunk 6]
```diff
@@ -1509,18 +1466,17 @@ class MyPatternProcessFunction<IN, OUT> extends PatternProcessFunction<IN, OUT>
 }
 {% endhighlight %}
 
-The `PatternProcessFunction` gives access to a `Context` object. Thanks to it, one can access time related
-characteristics such as `currentProcessingTime` or `timestamp` of current match (which is the timestamp of the last element assigned to the match).
-For more info see [Time context](#time-context).
-Through this context one can also emit results to a [side-output]({{ site.baseurl }}/dev/stream/side_output.html).
+`PatternProcessFunction`可以访问`Context`对象。有了它之后，你可以访问时间属性比如`currentProcessingTime`或者当前匹配的`timestamp`
```
Comment (dianfu) [inline docs/dev/libs/cep.zh.md]: 你可以访问时间属性 后面加个逗号？

File: docs/dev/libs/cep.zh.md [hunk 7]
```diff
@@ -1509,18 +1466,17 @@ class MyPatternProcessFunction<IN, OUT> extends PatternProcessFunction<IN, OUT>
...
+（最新分配到匹配上的事件的时间戳）.
```
Comment (dianfu) [inline docs/dev/libs/cep.zh.md]: . -> 。

File: docs/dev/libs/cep.zh.md [hunk 8]
```diff
@@ -1509,18 +1466,17 @@ class MyPatternProcessFunction<IN, OUT> extends PatternProcessFunction<IN, OUT>
...
-#### Handling Timed Out Partial Patterns
+#### 处理超时的部分匹配
 
-Whenever a pattern has a window length attached via the `within` keyword, it is possible that partial event sequences
-are discarded because they exceed the window length. To act upon a timed out partial match one can use `TimedOutPartialMatchHandler` interface.
-The interface is supposed to be used in a mixin style. This mean you can additionally implement this interface with your `PatternProcessFunction`.
-The `TimedOutPartialMatchHandler` provides the additional `processTimedOutMatch` method which will be called for every timed out partial match.
+当一个模式上通过`within`加上窗口长度后，部分匹配的事件序列就可能因为超过窗口长度而被丢弃。可以使用`TimedOutPartialMatchHandler`接口
+来处理超时的部分匹配。这个接口可以和其它的混合使用。也就是说你可以在自己的`PatternProcessFunction`里另外实现这个接口。
+`TimedOutPartialMatchHandler`提供了另外的`processTimedOutMatch`方法，这个方法对每个超时的部分匹配都对调用。
```
Comment (dianfu) [inline docs/dev/libs/cep.zh.md]: 都对调用 -> 都会调用

File: docs/dev/libs/cep.zh.md [hunk 9]
```diff
@@ -1598,17 +1554,17 @@ val timeoutResult: DataStream[TimeoutEvent] = result.getSideOutput(outputTag)
...
 
-## Time in CEP library
+## CEP库中的时间
 
-### Handling Lateness in Event Time
+### 按照事件时间处理晚到事件
 
-In `CEP` the order in which elements are processed matters. To guarantee that elements are processed in the correct order when working in event time, an incoming element is initially put in a buffer where elements are *sorted in ascending order based on their timestamp*, and when a watermark arrives, all the elements in this buffer with timestamps smaller than that of the watermark are processed. This implies that elements between watermarks are processed in event-time order.
+在`CEP`中，事件的处理顺序很重要。在使用事件时间时，为了保证事件按照正确的顺序被处理，一个事件到来后会先被放到一个缓冲区中，
+在缓冲区里事件都按照时间戳从小到大排序，当水位线到达后，缓冲区中所有小于水位线的事件被处理。这意味这水位线之间的数据都按照事件戳被顺序处理。
```
Comment (dianfu) [inline docs/dev/libs/cep.zh.md]: 事件戳 -> 时间戳

File: docs/dev/libs/cep.zh.md [hunk 10]
```diff
@@ -1750,31 +1704,25 @@ val alerts = patternStream.select(createAlert(_))
...
-2. Change your functions provided as arguments to the `select(...)` and `flatSelect(...)` methods to expect a list of
-events associated with each pattern (`List` in `Java`, `Iterable` in `Scala`). This is because with the addition of
-the looping patterns, multiple input events can match a single (looping) pattern.
+2. 修改你作为`select(...)`和`flatSelect(...)`方法的参数的函数为期望每个模式关联一个事件列表（`Java`中`List`，`Scala`中`Iterable`）。
+这是因为增加了循环模式后，多个事件可能匹配一个单一的（循环）模式。
 
-3. The `followedBy()` in Flink 1.1 and 1.2 implied `non-deterministic relaxed contiguity` (see
-[here](#conditions-on-contiguity)). In Flink 1.3 this has changed and `followedBy()` implies `relaxed contiguity`,
-while `followedByAny()` should be used if `non-deterministic relaxed contiguity` is required.
+3. 在Flink 1.1和1.2中，`followedBy()` in Flink 1.1 and 1.2隐含了`不确定的松散连续` (参见[这里](#组合模式))。
```
Comment (dianfu) [inline docs/dev/libs/cep.zh.md]: and -> 和

Comment (shuai-xu) [reply docs/dev/libs/cep.zh.md]: done

File: docs/dev/libs/cep.zh.md (hunk 9 above)
Comment (shuai-xu) [reply docs/dev/libs/cep.zh.md]: done

File: docs/dev/libs/cep.zh.md (hunk 8 above)
Comment (shuai-xu) [reply docs/dev/libs/cep.zh.md]: done

File: docs/dev/libs/cep.zh.md (hunk 7 above)
Comment (shuai-xu) [reply docs/dev/libs/cep.zh.md]: done

File: docs/dev/libs/cep.zh.md (hunk 6 above)
Comment (shuai-xu) [reply docs/dev/libs/cep.zh.md]: done

File: docs/dev/libs/cep.zh.md [hunk 11]
```diff
@@ -736,41 +715,39 @@ Pattern<Event, ?> relaxedNot = start.notFollowedBy("not").where(...);
...
 
-It's also possible to define a temporal constraint for the pattern to be valid.
-For example, you can define that a pattern should occur within 10 seconds via the `pattern.within()` method.
-Temporal patterns are supported for both [processing and event time]({{site.baseurl}}/dev/event_time.html).
+也可以为模式定义一个有效时间约束。
+例如，你可以通过`pattern.within()`方法指定一个模式应该在10秒内发生。
+这种暂时的模式支持[处理时间和事件时间]({{site.baseurl}}/zh/dev/event_time.html).
 
-{% warn Attention %} A pattern sequence can only have one temporal constraint. If multiple such constraints are defined on different individual patterns, then the smallest is applied.
+{% warn 注意 %} 一个模式序列只能有一个时间限制。如果在限制了多个时间在不同的单个模式上，会使用最小的那个时间限制。
```
[the remark this answers is not in the document]
Comment (shuai-xu) [reply docs/dev/libs/cep.zh.md]: done

File: docs/dev/libs/cep.zh.md [hunk 12]
```diff
@@ -136,140 +132,143 @@ val result: DataStream[Alert] = patternStream.process(
...
+指定判断事件属性的条件可以通过`pattern.where()`、`pattern.or()`或者`pattern.until()`方法。
+这些可以是`IterativeCondition`或者`SimpleCondition`。
 
-**Iterative Conditions:** This is the most general type of condition. This is how you can specify a condition that
-accepts subsequent events based on properties of the previously accepted events or a statistic over a subset of them.
+**迭代条件:** 这是最普遍的条件类型。使用它可以指定一个基于前面已经被接受的事件的属性或者它们的一个子集的统计数据来决定是否接受时间序列的条件。
 
-Below is the code for an iterative condition that accepts the next event for a pattern named "middle" if its name starts
-with "foo", and if the sum of the prices of the previously accepted events for that pattern plus the price of the current event do not exceed the value of 5.0. Iterative conditions can be powerful, especially in combination with looping patterns, e.g. `oneOrMore()`.
+下面是一个迭代条件的代码，它接受"middle"模式下一个事件的名称开头是"foo"， 并且前面已经匹配到的事件的加个加上这个事件的价格小于5.0。
```
[the remark this answers is not in the document]
Comment (shuai-xu) [reply docs/dev/libs/cep.zh.md]: done

File: docs/dev/libs/cep.zh.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (shuai-xu) [reply docs/dev/libs/cep.zh.md]: done

File: docs/dev/libs/cep.zh.md [hunk 13]
```diff
@@ -1651,44 +1607,42 @@ val lateData: DataStream[String] = result.getSideOutput(lateDataOutputTag)
...
 
-This context gives user access to time characteristics of processed events (incoming records in case of `IterativeCondition` and matches in case of `PatternProcessFunction`).
-Call to `TimeContext#currentProcessingTime` always gives you the value of current processing time and this call should be preferred to e.g. calling `System.currentTimeMillis()`.
+这个上下文让用户可以获取处理的事件（在`IterativeCondition`时候是进来的记录，在`PatternProcessFunction`是匹配的结果）的时间属性。
+调用`TimeContext#currentProcessingTime`总是返回当前的处理时间，而且尽量去调用这个函数而不是调用其它的比如说`System.currentTimeMillis()`。
 
-In case of `TimeContext#timestamp()` the returned value is equal to assigned timestamp in case of `EventTime`. In `ProcessingTime` this will equal to the point of time when said event entered
-cep operator (or when the match was generated in case of `PatternProcessFunction`). This means that the value will be consistent across multiple calls to that method.
+使用`EventTime`时，`TimeContext#timestamp()`返回的值等于分配的时间戳。
+使用`ProcessingTime`是，这个值等于事件进入CEP算子的时间点（在`PatternProcessFunction`中是匹配产生的时间）。
```
[the remark this answers is not in the document]
Comment (shuai-xu) [reply docs/dev/libs/cep.zh.md]: done

File: docs/dev/libs/cep.zh.md [hunk 14]
```diff
@@ -335,88 +332,87 @@ start.where(event => event.getName.startsWith("foo"))
...
 {% highlight scala %}
-start.subtype(classOf[SubEvent]).where(subEvent => ... /* some condition */)
+start.subtype(classOf[SubEvent]).where(subEvent => ... /* 一些判断条件 */)
 {% endhighlight %}
 </div>
 </div>
 
-**Combining Conditions:** As shown above, you can combine the `subtype` condition with additional conditions. This holds for every condition. You can arbitrarily combine conditions by sequentially calling `where()`. The final result will be the logical **AND** of the results of the individual conditions. To combine conditions using **OR**, you can use the `or()` method, as shown below.
+**组合条件：** 如上所示，你可以把`subtype`条件和其他的条件结合起来使用。这适用于任何条件，你可以通过依次调用`where()`来组合条件。
+最终的结果是是每个单一条件的结果的逻辑**AND**。如果想使用**OR**来组合条件，你可以使用像下面这样使用`or()`方法。
```
[the remark this answers is not in the document]
Comment (shuai-xu) [reply docs/dev/libs/cep.zh.md]: done

File: docs/dev/libs/cep.zh.md [hunk 15]
```diff
@@ -1537,14 +1493,14 @@ class MyPatternProcessFunction<IN, OUT> extends PatternProcessFunction<IN, OUT>
...
+但你可以通过`Context`对象把结果输出到[侧输出]({{ site.baseurl }}/zh/dev/stream/side_output.html)。
 
 
-#### Convenience API
+#### 便捷的API
 
-The aforementioned `PatternProcessFunction` was introduced in Flink 1.8 and since then it is the recommended way to interact with matches.
-One can still use the old style API like `select`/`flatSelect`, which internally will be translated into a `PatternProcessFunction`.
+前面提到的`PatternProcessFunction`是在Flink 1.8之后引入的，从那之后推荐使用这个接口来处理匹配到的结果。
+仍然可以使用像`select`/`flatSelect`这样旧格式的API，它们会在内部被转换为`PatternProcessFunction`。
```
[the remark this answers is not in the document]
Comment (shuai-xu) [reply docs/dev/libs/cep.zh.md]: done

File: docs/dev/libs/cep.zh.md (hunk 14 above)
[the remark this answers is not in the document]
Comment (shuai-xu) [reply docs/dev/libs/cep.zh.md]: done

File: docs/dev/libs/cep.zh.md [hunk 16]
```diff
@@ -425,18 +421,17 @@ pattern.where(new IterativeCondition<Event>() {
         <tr>
             <td><strong>or(condition)</strong></td>
             <td>
-                <p>Adds a new condition which is ORed with an existing one. An event can match the pattern only if it
-                passes at least one of the conditions:</p>
+                <p>增加一个新的判断和当前的判断取或。一个事件只要满足至少一个判断条件就匹配到模式：</p>
```
[the remark this answers is not in the document]
Comment (shuai-xu) [reply docs/dev/libs/cep.zh.md]: done

File: docs/dev/libs/cep.zh.md [hunk 17]
```diff
@@ -23,23 +23,20 @@ specific language governing permissions and limitations
...
+FlinkCEP是在Flink上层实现的复杂事件处理库。
+它可以让你在无限事件流中检测出特定的事件模型，有机会掌握数据中重要的那部分。
 
-This page describes the API calls available in Flink CEP. We start by presenting the [Pattern API](#the-pattern-api),
-which allows you to specify the patterns that you want to detect in your stream, before presenting how you can
-[detect and act upon matching event sequences](#detecting-patterns). We then present the assumptions the CEP
-library makes when [dealing with lateness](#handling-lateness-in-event-time) in event time and how you can
-[migrate your job](#migrating-from-an-older-flink-versionpre-13) from an older Flink version to Flink-1.3.
+本页讲述了Flink CEP中可用的API，我们首先讲述[模式API](#模式api)，它可以让你指定想在数据流中检测的模式，然后讲述如何[检测匹配的事件序列并进行处理](#检测模式)。
+再然后我们讲述Flink在按照事件时间[处理迟到事件](#按照事件时间处理晚到事件)时的假设，
```
[the remark this answers is not in the document]
Comment (shuai-xu) [reply docs/dev/libs/cep.zh.md]: done

File: docs/dev/libs/cep.zh.md [hunk 18]
```diff
@@ -445,15 +440,14 @@ pattern.where(new IterativeCondition<Event>() {
               <tr>
                  <td><strong>until(condition)</strong></td>
                  <td>
-                     <p>Specifies a stop condition for a looping pattern. Meaning if event matching the given condition occurs, no more
-                     events will be accepted into the pattern.</p>
-                     <p>Applicable only in conjunction with <code>oneOrMore()</code></p>
-                     <p><b>NOTE:</b> It allows for cleaning state for corresponding pattern on event-based condition.</p>
```
[the remark this answers is not in the document]
Comment (shuai-xu) [reply docs/dev/libs/cep.zh.md]: 叫提示更好一点吧

[the remark this answers is not in the document]
Comment (dianfu) [reply docs/dev/libs/cep.zh.md]: 嗯，是的

File: docs/dev/libs/cep.zh.md [hunk 19]
```diff
@@ -736,41 +715,39 @@ Pattern<Event, ?> relaxedNot = start.notFollowedBy("not").where(...);
...
 
-3. Non-Deterministic Relaxed Contiguity between `"a"` and `"b"`: `{a b1}`, `{a b2}`, as this is the most general form.
+3. `"a"`和`"b"`之间不确定的松散连续： `{a b1}`, `{a b2}`，这是最常见的情况。
 
-It's also possible to define a temporal constraint for the pattern to be valid.
-For example, you can define that a pattern should occur within 10 seconds via the `pattern.within()` method.
-Temporal patterns are supported for both [processing and event time]({{site.baseurl}}/dev/event_time.html).
+也可以为模式定义一个有效时间约束。
+例如，你可以通过`pattern.within()`方法指定一个模式应该在10秒内发生。
+这种暂时的模式支持[处理时间和事件时间]({{site.baseurl}}/zh/dev/event_time.html).
```
[the remark this answers is not in the document]
Comment (shuai-xu) [reply docs/dev/libs/cep.zh.md]: 叫时间模式吧，这个好像合适一点

**Comment by dianfu** [comment]
@shuai-xu Thanks for the update. Merging...

**Pull request merged**